tests/util/test_executable_runner.py

```diff
@@ -29,22 +30,43 @@ def test_validate_executable_path_not_executable() -> None:
             ExecutableRunner.validate_executable_path(executable=tmpfile.name)
 
 
-def test_validate_executable_path() -> None:
-    exec = "yes"
-    exec_full_str = f"/usr/bin/{exec}"
-    exec_full_path = Path(exec_full_str)
-    assert exec_full_path.is_absolute()
+@pytest.mark.parametrize("executable", ["/usr/bin/yes", Path("/usr/bin/yes")])
+def test_validate_executable_path_returns_existing_paths(executable: str | Path) -> None:
+    """
+    `validate_executable_path` should return the `yes` executable when the absolute path to the
+    executable is passed, either as a string or a Path.
+    """
+    expected_path: Path = Path("/usr/bin/yes")
 
-    # find it on the PATH
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec)
-    # find it given an absolute path as a string
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec_full_str)
-    # find it given an absolute path as a Path
-    assert exec_full_path == ExecutableRunner.validate_executable_path(executable=exec_full_path)
+    with mock.patch.dict(os.environ):
+        # Clear the PATH, in case the user has a local version of `yes` elsewhere on their PATH
+        os.environ.pop("PATH")
 
-    # do not find it on the PATH if given as a Path
+        validated_path: Path = ExecutableRunner.validate_executable_path(executable=executable)
+        assert validated_path == expected_path
+
+
+def test_validate_executable_finds_on_path() -> None:
+    """
+    `validate_executable_path` should find executables on the PATH when the name of the executable
+    is passed as a string.
+    """
+    expected_path: Path = Path("/usr/bin/yes")
+
+    with mock.patch("shutil.which", return_value=expected_path.as_posix()):
```

Comment:
NB: I briefly tried monkeying around with patching `os.environ["PATH"]`, but `shutil.which` was still returning results based on the value of the PATH at the time 

I think the mock here is sufficient